### ISSUE TITLE
The documentation on TLS secrets can be misleading

### ISSUE THREAD

**Issue by 123MwanjeMike** [kubernetes/website#41785]
**This is a Bug Report**

**Problem:**
The documentation and example on the TLS secrets can be misleading to a new user.
The documentation does not mention anywhere that the values for  both `data.tls.key` and `data.tls.key` should be base64 encoded and yet any workloads using such a secret will automatically decode the value hence causing errors such as 
```
failed to find any PEM data in certificate input
```
The example yaml provided does not help much either given that the sample values(`MIIC2DCCAcCgAwIBAgIBATANBgkqh` and `MIIEpgIBAAKCAQEA7yn3bRHQ5FHMQ`) are those before encoding.

**Proposed Solution:**
The documentation and example need to be updated to reflect that the `data.tls.key` and `data.tls.key` values should be base64 encoded

**Page to Update:**
https://kubernetes.io/docs/concepts/configuration/secret/

Kubernetes Version:
1.26.4

Additional Information:
- Sample error log
  ```
  I0627 18:42:34.500254       1 main.go:234] Kubernetes version: 1.26.4
  I0627 18:42:34.507952       1 main.go:380] Using nginx version: nginx/1.23.4
  I0627 18:42:34.525283       1 main.go:776] Pod label updated: nginx-ingress-dev-controller-p6jjp
  F0627 18:42:34.535386       1 main.go:413] Error trying to get the default server TLS secret nginx-ingress/tls-secret: nginx-ingress/tls-secret is invalid: Failed to validate TLS cert and key: tls: failed to find any PEM data in certificate input
  ```

**Comment by sftim** [comment]
This issue duplicates https://github.com/kubernetes/website/issues/32869

/triage duplicate

**Comment by 123MwanjeMike** [comment]
Fixed by #41786
